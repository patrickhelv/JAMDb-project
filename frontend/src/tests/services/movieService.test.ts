import MovieService from '../../app/services/movieService'

describe('Movie services', () => {
    const movieService = new MovieService()
    const movieDetail = {
        _id: '1',
        title: 'Test movie',
        description: 'Test description',
        published: 2000,
    }
    const { _id, title, description, published } = movieDetail

    it('Get movies', () => {
        expect(movieService.getMovies()).toMatchSnapshot()
    })

    it('Find movie', () => {
        expect(movieService.findMovie(movieDetail._id)).toMatchSnapshot()
    })

    it('Create movie and update movie', async () => {
        // TODO
        // const response = await movieService.createMovie(title, description, published)
        // expect(response).toMatchSnapshot()
        // const movieResult = await movieService.searchMovie(response.title)
        // const movie = movieResult[0]
        // expect(response).toBeInstanceOf(movie)
        // expect(movieService.updateMovie(movie._id, title, description, published)).toMatchSnapshot()
    })
})
