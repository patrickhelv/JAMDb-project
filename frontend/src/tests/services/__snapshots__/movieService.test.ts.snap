// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie services Find movie 1`] = `Promise {}`;
